profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestEntitlement.java

```diff
@@ -2049,4 +2053,61 @@ public Boolean call() throws Exception {
         });
     }
 
+    @Test(groups = "slow")
+    public void testCreateSubscriptionWithOnlyUsageOverrides() throws Exception {
+        final LocalDate initialDate = new LocalDate(2012, 4, 25);
+        clock.setDay(initialDate);
+
+        final Account accountJson = createAccountWithDefaultPaymentMethod();
+
+        final Subscription input = new Subscription();
+        input.setAccountId(accountJson.getAccountId());
+        input.setPlanName("trebuchet-usage-in-arrear");
+
+        final PhasePrice recurringOverride = new PhasePrice();
+        recurringOverride.setPlanName("trebuchet-usage-in-arrear");
+        recurringOverride.setPhaseType("EVERGREEN");
+
+        final UsagePrice usagePrice = new UsagePrice();
+        usagePrice.setUsageName("u1");
+        usagePrice.setUsageType(UsageType.CAPACITY);
+
+        final List<TierPrice> tierPrices = new ArrayList<>();
+
+        BlockPrice blockPrice = new BlockPrice();
+        blockPrice.setPrice(new BigDecimal("120"));
+        blockPrice.setMax(new BigDecimal(100));
+        blockPrice.setUnitName("stones");
+        TierPrice tierPrice = new TierPrice();
+        tierPrice.setBlockPrices(List.of(blockPrice));
+        tierPrices.add(tierPrice);
+
+        blockPrice = new BlockPrice();
+        blockPrice.setPrice(new BigDecimal("1100"));
+        blockPrice.setMax(new BigDecimal(-1));
+        blockPrice.setUnitName("stones");
+        tierPrice = new TierPrice();
+        tierPrice.setBlockPrices(List.of(blockPrice));
+        tierPrices.add(tierPrice);
+
+        usagePrice.setTierPrices(tierPrices);
+
+        recurringOverride.setUsagePrices(List.of(usagePrice));
+        input.setPriceOverrides(List.of(recurringOverride));
+
+        callbackServlet.pushExpectedEvents(ExtBusEventType.ACCOUNT_CHANGE,
+                                           ExtBusEventType.ENTITLEMENT_CREATION,
+                                           ExtBusEventType.SUBSCRIPTION_CREATION,
+                                           ExtBusEventType.SUBSCRIPTION_CREATION);
+
+        final Subscription entitlementJson = subscriptionApi.createSubscription(input,
+                                                                                null,
+                                                                                (LocalDate) null,
+                                                                                NULL_PLUGIN_PROPERTIES,
+                                                                                requestOptions);
+        callbackServlet.assertListenerStatus();
+        final Subscription subscription = subscriptionApi.getSubscription(entitlementJson.getSubscriptionId(), requestOptions);
+        Assert.assertEquals(subscription.getState(), EntitlementState.ACTIVE);
```

Comment:
Any way we could check that price was correctly overridden - would `subscription#getLastActivePlan()` provide such info?

Reply:
Addressed in 984a82d